libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -199,7 +199,17 @@ void AP_Motors::add_motor_num(int8_t motor_num)
     if (motor_num >= 0 && motor_num < AP_MOTORS_MAX_NUM_MOTORS) {
         uint8_t chan;
         SRV_Channel::Aux_servo_function_t function = SRV_Channels::get_motor_function(motor_num);
-        SRV_Channels::set_aux_channel_default(function, motor_num);
+#if HAL_WITH_IO_MCU
+        // can't do digital outputs with IOMCU, so remap to AUX if possible
+        if (is_digital_pwm_type() && motor_num < 6) {
+            SRV_Channels::set_aux_channel_default(function, motor_num + 8);
+        } else {
+#endif
+            SRV_Channels::set_aux_channel_default(function, motor_num);
+#if HAL_WITH_IO_MCU
+        }
+#endif
+
         if (!SRV_Channels::find_channel(function, chan)) {
             gcs().send_text(MAV_SEVERITY_ERROR, "Motors: no SERVOx_FUNCTION set to Motor%u", motor_num + 1);
         }
```

Comment:
Slighly unrelated, but this warning is pointless because it happens before we have GCS connection.